apps/content-publishing-worker/src/publisher/message.publisher.ts

```diff
@@ -32,6 +34,7 @@ export class MessagePublisher implements OnApplicationBootstrap {
   async onApplicationBootstrap() {
     // Get the real batch size when the application starts
     this.maxBatchSize = await this.blockchainRpcService.maximumCapacityBatchLength();
+    this.batchProcessWindowMs = this.blockchainService.blockTimeMs;
```

Comment:
Just need to be careful doing this in the bootstrap code; might need to await the api--the api consts are fetched from the chain when it first connects (though there is a hard-coded fallback value of 6000ms in our code)

Reply:
Ah interesting - I noticed that the blockTimeMs is synchronous, so I didn't await it. But I am seeing that there is a isReady async function that checks to see if the api is ready, I can await that and then grab the blockTimeMs.

Reply:
Since the block time is almost never going to change, why get it from the chain?  It's simpler just to set a constant, and then you don't have to worry about any of the above.

Reply:
I'm just concerned that we've got a proliferation of "magic" values that, at the end of the day, are predicated on 6-second block times, so why not make that relationship more obvious?

We actually used to have a constant `BLOCK_TIME_MS` or something like that (env var), but once I realized I could get that value from the chain I eliminated it.